src/difference/difference_solution1/difference.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * This function creates an array of values that are in the first array,
+ * but not not in the second array.
+ *
+ * Repeated values are not duplicated in the return value,
+ * and the order of result values are determined by the first array.
+ *
+ * **Note:** This function returns a new array, and has no side-effects.
+ *
+ * @param {Array} [arrayToFindTheDifference=[]] - The array to inspect.
+ * @param {Array} [arrayToCompare=[]] - The values to exclude.
+ *
+ * @returns {Array} Returns the new array of filtered values.
+ *
+ * @example
+ *
+ * difference([2, 1], [2, 3]); //  [1]
+ *
+ * @example
+ *
+ * difference([1, 2, 1], [2, 3]); //  [1]
+ */
+export const arrayDifference = (
+  arrayToFindTheDifference = [],
+  arrayToCompare = [],
+) => {
+  // ensure that a new array is created
+  const returnArray = [];
+  // ensure that every element in the array is checked
+  for (let i = 0; i < arrayToFindTheDifference.length; i++) {
```

Comment:
When you are going to iterate only on elements, and you don't need the indexes a `for of loop` is more compact and more easy to read